api/proto/teleport/gitserver/v1/git_server_service.proto

```diff
@@ -0,0 +1,86 @@
+// Copyright 2024 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.gitserver.v1;
+
+import "google/protobuf/empty.proto";
+import "teleport/legacy/types/types.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/gitserver/v1;gitserverv1";
+
+// GitServerService provides methods to manage Git server.
+service GitServerService {
+  // CreateGitServer is used to create a Git server object.
+  rpc CreateGitServer(CreateGitServerRequest) returns (types.ServerV2);
+  // GetGitServer is used to retrieve a Git server object.
+  rpc GetGitServer(GetGitServerRequest) returns (types.ServerV2);
+  // ListGitServers is used to query Git servers.
+  rpc ListGitServers(ListGitServersRequest) returns (ListGitServersResponse);
+  // UpdateGitServer is used to update a Git server object.
+  rpc UpdateGitServer(UpdateGitServerRequest) returns (types.ServerV2);
+  // UpsertGitServer is used to create or replace a Git server object.
+  rpc UpsertGitServer(UpsertGitServerRequest) returns (types.ServerV2);
+  // DeleteGitServer is used to delete a Git server object.
+  rpc DeleteGitServer(DeleteGitServerRequest) returns (google.protobuf.Empty);
+}
+
+// CreateGitServerRequest is a request to create a Git server.
+message CreateGitServerRequest {
+  // Server is the Git server to create.
+  types.ServerV2 server = 1;
```

Comment:
Theoretical question :-) 
If there any possible way to not mix gogole proto with legacy deprecated gogo proto ?

Reply:
it's probably doable. i will have to do conversions back AND force in a LOT of places as git_server will rely on existing SSH transport. I prefer use the old type and keep backend consistent as the other server resources.

Reply:
Is `types.ServerV2` leveraged for literally anything here? Is it planned to work for arbitrary git servers reached directly from the server?

Reply:
> Is it planned to work for arbitrary git servers reached directly from the server?

What do you mean? 

Do you mean directly from the proxy server? there will be git agents eventually for private/self-hosted git services.

Reply:
Well, the entire point of ServerV2 is that it has a bunch of fields that make sense for servers - so far, a `git_server` with subkind `github` is literally only using metadata and one field that was added to ServerSpecV2 that only makes sense in that exact scenario, so I was wondering if in the future we're planning to make use of ServerV2.